release/src/Main.hs

```diff
@@ -50,7 +50,26 @@ main = do
       let nonDeployJars = filter (not . isDeployJar . artReleaseType) mavenUploadArtifacts
       let allMavenTargets = Set.fromList $ fmap (T.unpack . getBazelTarget . artTarget) mavenUploadArtifacts
 
-      -- first find out all the missing internal dependencies
+      -- check that all maven artifacts use com.daml as groupId
+      let nonComDamlGroupId = filter (\a -> ("com.daml" /=) $ foldr (<>) "" (List.intersperse "." $ map T.unpack $ pomGroupId $ artMetadata a)) mavenUploadArtifacts
+      when (not (null nonComDamlGroupId)) $ do
+          $logError "Some artifacts don't use com.daml as groupId!"
+          forM_ nonComDamlGroupId $ \artifact -> do
+              $logError ("\t- "# getBazelTarget (artTarget artifact))
+          liftIO exitFailure
+
+      -- check that no artifact id is used more than once
+      let groupedArtifacts = List.groupBy (\a b -> (pomArtifactId $ artMetadata a) == (pomArtifactId $ artMetadata b)) mavenUploadArtifacts
```

Comment:
```suggestion
      let groupedArtifacts = List.groupOn (pomArtifactId . artMetadata) mavenUploadArtifacts
```
Needs a change in the import from `Data.List` to `Data.List.Extra` but we already depend on `extra` anyway so no reason not to use it.

Reply:
Thank you. This was annoying me as well, but didn't know about `groupOn`.